configuration/config2/configfiles/types.xsd

```diff
@@ -0,0 +1,65 @@
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+################################################################################
+ -->
+<xs:schema 
+    xmlns:xs="http://www.w3.org/2001/XMLSchema" elementFormDefault="qualified" attributeFormDefault="unqualified" 
+    xmlns:hpcc="someurl">
+    <xs:simpleType name="percent">
+        <xs:restriction base="xs:integer">
+            <xs:minInclusive value="0"/>
+            <xs:maxInclusive value="100"/>
+        </xs:restriction>
+    </xs:simpleType>
+    <xs:simpleType name="nodeName">
+        <xs:restriction base="xs:string">
+            <xs:pattern value="[a-zA-z0-9_\-]+"/>
+            <!-- need regex to eliminate special characters -->
+        </xs:restriction>
+    </xs:simpleType>
+    <xs:simpleType name="ipV4Address">
+        <xs:restriction base="xs:token">
+            <xs:pattern value="/^0*([1-9]?\\d|1\\d\\d|2[0-4]\\d|25[0-5])\\.0*([1-9]?\\d|1\\d\\d|2[0-4]\\d|25[0-5])\\.0*([1-9]?\\d|1\\d\\d|2[0-4]\\d|25[0-5])\\.0*([1-9]?\\d|1\\d\\d|2[0-4]\\d|25[0-5])$/"/>
+        </xs:restriction>
+    </xs:simpleType>
+	<xs:simpleType name="absolutePath">
+		<xs:restriction base="xs:string">
+			<xs:pattern value="/^\\/.*/"/>
+		</xs:restriction>
+	</xs:simpleType>
+	<xs:simpleType name="AutoTimeStampType">
+		<xs:restriction base="xs:string" hpcc:autoType="timestamp"/>
+	</xs:simpleType>
+	<xs:simpleType name="AutoComputerType">
+		<xs:restriction base="xs:string" hpcc:autoType="requestIP"/>
+	</xs:simpleType>
+	<xs:simpleType name="AutoUseridType">
+		<xs:restriction base="xs:string" hpcc:autoType="requestUser"/>
+	</xs:simpleType>
+	<xs:attributeGroup name="buildInfo">
+		<xs:attribute name="build" type="xs:string" use="required" hpcc:hidden="true" hpcc:tooltip="The build name to be deployed"/>
+        <xs:attribute name="buildSet" type="xs:string" use="required" hpcc:hidden="true"/>
+	</xs:attributeGroup>
+    <xs:attributeGroup name="computerNode">
+        <xs:attribute name="name" type="nodeName" use="required"/>
+        <xs:attribute name="netAddress" type="ipV4Address" use="required"/>
+    </xs:attributeGroup>
+	<xs:attributeGroup name="computerNodeReference">
+        <xs:attribute name="computer" type="nodeName" use="required" hpcc:readOnly="true" hpcc:mirrorFrom="/Environment/Hardware/Computer@name"/>
+        <xs:attribute name="netAddress" type="ipV4Address" use="required" hpcc:readOnly="true" hpcc:mirrorFrom="/Environment/Hardware/Computer@netAddress"/>
```

Comment:
I'm not sure how relevant this is, but configmanager has to allow ip addresses and hostnames in all netaddress type fields.

Reply:
The computerNodeReference attribute group is for use by any component that needs to list computer instances to which their component is bound. It marks the nodeName and ip address as readonly so that they can only view these values for inspection only. The source, in the Hardware component, is the only place where these values can be changed. the mirrorFrom keyword ensures that when a value is changed in the Hardware component, the value gets replicated to any component using this attribute group.

Reply:
ok so are you allowing hostnames (not to be confused with the name used to reference a computer node within confmgr)? It doesn't look like we do, please make sure we do.